### PR TITLE
openstack fail early if the root CA is malformed

### PR DESCRIPTION
**Pull request by eslutsky**
Dont fall back to system-CA cert pool.

### CODE REVIEW

File: pkg/controller/provider/container/openstack/client.go [hunk 1]
```diff
@@ -126,15 +126,20 @@ func (r *Client) Authenticate() (err error) {
 		} else {
 			cacert := []byte(r.getStringFromSecret(CACert))
 			roots := x509.NewCertPool()
-			ok := roots.AppendCertsFromPEM(cacert)
-			if !ok {
-				r.Log.Info("the CA certificate is malformed or was not provided, falling back to system CA cert pool")
+
+			if len(cacert) == 0 {
```
Comment (mmartinv) [inline pkg/controller/provider/container/openstack/client.go:130]: From https://pkg.go.dev/crypto/tls#Config:
>  // RootCAs defines the set of root certificate authorities
>	// that clients use when verifying server certificates.
>	// If RootCAs is nil, TLS uses the host's root CA set.
>	RootCAs *[x509](https://pkg.go.dev/crypto/x509).[CertPool](https://pkg.go.dev/crypto/x509#CertPool)

So maybe we can simplify this block.

File: pkg/controller/provider/container/openstack/client.go [hunk 2]
```diff
@@ -126,15 +126,20 @@ func (r *Client) Authenticate() (err error) {
...
+				r.Log.Info("CA certificate was not provided, falling back to system CA cert pool")
```
Comment (ahadas) [inline pkg/controller/provider/container/openstack/client.go:131]: then it's not a fall back, how about
```suggestion
				r.Log.Info("CA certificate was not provided, system CA cert pool is used")
```

File: pkg/controller/provider/container/openstack/client.go [hunk 3]
```diff
@@ -126,15 +126,20 @@ func (r *Client) Authenticate() (err error) {
...
 				roots, err = x509.SystemCertPool()
 				if err != nil {
 					err = liberr.New("failed to configure the system's CA cert pool")
 					return
 				}
 			}
+
+			ok := roots.AppendCertsFromPEM(cacert)
```
Comment (ahadas) [inline pkg/controller/provider/container/openstack/client.go:139]: in an else-block?

Comment (eslutsky) [reply pkg/controller/provider/container/openstack/client.go]: moved to else block

File: pkg/controller/provider/container/openstack/client.go (hunk 2 above)
Comment (eslutsky) [reply pkg/controller/provider/container/openstack/client.go]: done

**Review by mmartinv — APPROVED** [review]
lgtm

**Pull request merged**